File: app/src/main/java/com/ianarbuckle/dublinbushelper/authentication/AuthPagerActivity.java
package com.ianarbuckle.dublinbushelper.authentication;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;

import com.ianarbuckle.dublinbushelper.BaseActivity;
import com.ianarbuckle.dublinbushelper.R;
import com.ianarbuckle.dublinbushelper.authentication.login.LoginFragment;
import com.ianarbuckle.dublinbushelper.authentication.register.RegisterFragment;

import butterknife.BindView;

/**
 * Created by Ian Arbuckle on 04/03/2017.
 *
 */

public class AuthPagerActivity extends BaseActivity {

  @BindView(R.id.tabs)
  TabLayout tabLayout;

  @BindView(R.id.viewpager)
  ViewPager viewPager;

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    initTabLayout();
    initPager();
  }

  @Override
  protected void initLayout() {
    setContentView(R.layout.activity_pager_auth);
  }

  private void initTabLayout() {
    tabLayout.addTab(tabLayout.newTab().setText("Sign In"));
    tabLayout.addTab(tabLayout.newTab().setText("Sign Up"));
  }

  private void initPager() {
    final AuthPagerAdapter adapter = new AuthPagerAdapter(getSupportFragmentManager(), tabLayout.getTabCount());
    viewPager.setAdapter(adapter);
    viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
    tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
      @Override
      public void onTabSelected(TabLayout.Tab tab) {
        viewPager.setCurrentItem(tab.getPosition());
      }

      @Override
      public void onTabUnselected(TabLayout.Tab tab) {

      }

      @Override
      public void onTabReselected(TabLayout.Tab tab) {

      }
    });

  }

  private class AuthPagerAdapter extends FragmentStatePagerAdapter {

    int numOfTabs;

    public AuthPagerAdapter(FragmentManager fragmentManager, int numOfTabs) {
      super(fragmentManager);
      this.numOfTabs = numOfTabs;
    }

    @Override
    public Fragment getItem(int position) {
      switch (position) {
        case 0:
          return LoginFragment.newInstance();
        case 1:
          return RegisterFragment.newInstance();
        default:
          return null;
      }
    }

    @Override
    public int getCount() {
      return numOfTabs;
    }
  }

}

File: app/src/main/java/com/ianarbuckle/dublinbushelper/authentication/AuthPagerActivity.kt
package com.ianarbuckle.dublinbushelper.authentication

import android.os.Bundle
import android.os.PersistableBundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import com.ianarbuckle.dublinbushelper.BaseActivity

import com.ianarbuckle.dublinbushelper.R
import com.ianarbuckle.dublinbushelper.authentication.login.LoginFragment
import com.ianarbuckle.dublinbushelper.authentication.register.RegisterFragment
import com.ianarbuckle.dublinbushelper.utils.ToolbarManager
import kotlinx.android.synthetic.main.layout_pager.*
import kotlinx.android.synthetic.main.layout_tab.*
import org.jetbrains.anko.find

/**
 * Created by Ian Arbuckle on 04/03/2017.

 */

class AuthPagerActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initToolbar(false)
        initTabLayout()
        initPager()
    }

    override fun initToolbar(isHomeAsUpEnabled: Boolean) {
        super.initToolbar(isHomeAsUpEnabled)
        toolbar.title = getString(R.string.app_name)
    }

    override fun initLayout() {
        setContentView(R.layout.activity_pager_auth)
    }

    private fun initTabLayout() {
        tabLayout.addTab(tabLayout.newTab().setText(R.string.login))
        tabLayout.addTab(tabLayout.newTab().setText(R.string.register))
    }

    private fun initPager() {
        val adapter = AuthPagerAdapter(supportFragmentManager, tabLayout.tabCount)
        viewPager.adapter = adapter
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))
        tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                viewPager.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {

            }

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
    }

    private inner class AuthPagerAdapter(fragmentManager: FragmentManager, internal var numOfTabs: Int) : FragmentStatePagerAdapter(fragmentManager) {

        override fun getItem(position: Int): Fragment? {
            when (position) {
                0 -> return LoginFragment.newInstance()
                1 -> return RegisterFragment.newInstance()
                else -> return null
            }
        }

        override fun getCount(): Int {
            return numOfTabs
        }
    }

}
